src/ceph.c

```diff
@@ -507,6 +485,23 @@ static _Bool has_suffix(char const *str, char const *suffix) {
   return 0;
 }
 
+static int cut_suffix(char new_str[], size_t new_str_len, char const *str,
```

Comment:
1.  The most common name for the first two arguments are `buffer` and `buffer_size`.
2.  Please use a `char *` instead of a `char []`. (Both are correct and mean essentially the same thing, but the latter is very uncommon, at least within the C codebase).
3.  You only ever return the value zero and you never check the return value. Make this a `void` function.

```c
static void cut_suffix(char *buffer, size_t buffer_size, ...)
```